e2e/tests/node_not_ready/node_power_off.robot

```diff
@@ -0,0 +1,370 @@
+    And Replica state on node 1 should be running
+    And Replica state on node 2 should be stopped
+    And Data should be intact
+
+    When Power on node 2
+
+    Then Node 2 state should be Ready
+    And Volume state should be healthy
+    And Engine state should be running
+    And Replica state on node 1 should be running
+    And Replica state on node 2 should be stopped
+    And Data should be intact
+    [Teardown]    Teardown
+
+Node power off timeout with replica-VA
+    [Documentation]    volume with replica on attached node and powering off the volume attached node
+    [Tags]    nodepoweringoff    3replicas    attachreplicasamenode    abnormalreplicanode    abnormalattachednode
+    # Recover: After restart the node, volume should be detached and replica is failed-Need manual to do reattached.
+    Given A ${volume_size_gb} GB ${volume_type} volume with 3 replicas
+    And Attach volume to node 1
+    And Write data into mount point
+
+    When Power off node 1
+    And Wait ${pod_eviction_time_out} seconds
+
+    Then Node 1 should have 1 volume replica
+    And Node 1 state should be NotReady
+    And Volume state should be unknown
+    And Engine state should be unknown
+    And Replica state on node 1 should be unknown
+
+    When Power on node 1
+
+    Then Node 1 state should be Ready
+    And Volume state should be healthy
+    And Engine state should be running
+    And All replicas state should be running
+    [Teardown]    Teardown
+
+Node power off timeout with replica
+    [Tags]    nodepoweringoff    3replicas    attachreplicadiffnode    abnormalreplicanode    abnormalnoattachednode
+    # Recover: After restart the node, delete the stopped replica, and re-update the replica count to do rebuilding
+    Given A ${volume_size_gb} GB ${volume_type} volume with 3 replicas
+    And Attach volume to node 1
+    And Write data into mount point
+
+    When Power off node 2
+    And Wait ${pod_eviction_time_out} seconds
+
+    Then Node 1 should have 1 volume replica
+    And Node 2 should have 1 volume replica
+    And Node 2 state should be NotReady
+    And Volume state should be degraded
+    And Engine state should be running
+    And Replica state on node 1 should be running
+    And Replica state on node 2 should be stopped
+    And Replica state on node 3 should be running
+
+    When Power on node 2
+
+    Then Node 2 state should be Ready
+    And Volume state should be healthy
+    And Engine state should be running
+    And All replicas state should be running
+    And Data should be intact
+    [Teardown]    Teardown
+
+Node power off timeout with VA
+    [Tags]    nodepoweringoff    2replicas    attachreplicasamenode    abnormalnoreplicanode    abnormalattachednode
+    Given A ${volume_size_gb} GB ${volume_type} volume with 2 replicas
+    And Attach volume to node 3
+    And Write data into mount point
+
+    When Power off node 3
+    And Wait ${pod_eviction_time_out} seconds
```

Comment:
Currently, it indeed won't wait until the Pod is evicted but only try to wait a configured long time to after the node powering off. Therefore, I updated the parameter name to `sleep_interval` to avoid confusion.

Reply:
Do the following steps require pods being evicted? Or what is the reason that we have to wait here?

Reply:
Yes, after a period of time when the node is powered off and pods are evicted, it checks the status of the node and all workloads. So I do this by sleeping until the POD eviction times out (5 minutes by default).
![image](https://github.com/longhorn/longhorn-tests/assets/17548901/9b9d46d8-a460-4017-92c2-e447fae4e0a3)

Reply:
If the following step requires pods being evicted before the step can be executed, it's better to implement the logic to exactly wait until pods being evicted. Or at least have a comment so we know it should be improved later.

Otherwise randomly using lots of `sleep` in the test code will eventually cause it meaninglessly time-consuming.

Reply:
I've add new keywords `Waiting for pods on node ${node_index} to be evicted` instead of `sleep`. 
https://github.com/longhorn/longhorn-tests/blob/2a2eadefdf2b61413e1e292a27010d9507ae1209/e2e/keywords/node.resource#L51-L53

I also ran a test to make sure that non-DeamonSet pods would be evicted if the node came back after the pod eviction timeout (default = 300 seconds).
![image](https://github.com/longhorn/longhorn-tests/assets/17548901/0a783080-6b26-468c-8afe-5cbf5e0e9d5f)